petab/v2/lint.py

```diff
@@ -394,6 +397,28 @@ def run(self, problem: Problem) -> ValidationIssue | None:
                 f"Condition table contains invalid targets: {invalid}"
             )
 
+        # Check that changes of simultaneously applied conditions don't
+        #  intersect
+        for experiment in problem.experiment_table.experiments:
+            for period in experiment.periods:
+                if not period.condition_ids:
+                    continue
+                period_targets = set()
+                for condition_id in period.condition_ids:
+                    condition_targets = {
+                        change.target_id
+                        for cond in problem.condition_table.conditions
+                        if cond.id == condition_id
+                        for change in cond.changes
+                    }
```

Comment:
Is this useful for tools? I guess tools might like some `problem.condition_table.get_updates(condition_id)` that provides a dictionary of updates to be applied. Then the keys can be used here.

Reply:
If it's already in a dict, the keys will be unique and we can't check for duplicates any. It works if we do the checking every time in such a function.
I think something like that will come when progressing with the importer.

Reply:
Not that it simplifies the code here, just that it might be useful to tools. But fine to keep as is until a tool dev requests something...